### PR TITLE
jRuby 9.x depends on java 8

### LINKED ISSUE #1313: JRuby 9.2+ requires Java 8, not 7
Minor issue...I noticed the definitions for JRuby 9k (9.x.x.x) all have "java7" as a required prerequisite, but since JRuby 9.2 we have required Java 8. The definitions should be tweaked.

### PR DESCRIPTION
**Pull request by afromankenobi**
Fix #1313

### CODE REVIEW

File: share/ruby-build/jruby-9.1.9.0 [hunk 1]
```diff
@@ -1,2 +1,2 @@
-require_java7
+require_java8
```
Comment (mislav) [inline share/ruby-build/jruby-9.1.9.0]: Can we leave 9.0 and 9.1 definitions intact, and just update jruby 9.2+ definitions?

**Comment by afromankenobi** [comment]
Yeah of course. Question, why?

File: share/ruby-build/jruby-9.1.9.0 (hunk 1 above)
Comment (afromankenobi) [reply share/ruby-build/jruby-9.1.9.0]: Updated. Please confirm! :)

**Comment by mislav** [comment]
Heads up: this broke things https://github.com/rbenv/ruby-build/issues/1373

I will fix it, but for the future, it's always a good idea to test your changes before you open pull requests. Thank you!

**Comment by afromankenobi** [comment]
Oh! So sorry. I just created the method for test locally but forgot to commit. Thanks for the correction

**Comment by mislav** [comment]
No worries! Ideally, our test suite should have caught things like these, but right now it doesn't source any definitions to check if all the directives make sense. Ideally, the test suite should "run" all definitions, but just stub out the download/configure/make part so it doesn't actually happen.

**Pull request merged**